opal/runtime/opal_progress.c

```diff
@@ -229,6 +236,15 @@ opal_progress(void)
         }
     }
 
+    if (opal_progress_poll_when_idle) {
+        if (events <= 0) {
+            if (yield_count++ > opal_progress_poll_threshold) {
+                poll(NULL, 0, 1);
+            }
+        } else {
+            yield_count = 0;
+        }
+    }
```

Comment:
If, per the comment above, we end up squashing both mechanisms into the same MCA variable, you should probably squash these two blocks together (i.e., the `yield()` and `poll()` blocks) so that we only have to have a single `if` statement (probably should be with an `OPAL_UNLIKELY`) gating entrance into that block.